### PR TITLE
Stop names display properly while creating new connection.

### PR DESCRIPTION
**Pull request by luktar**

### CODE REVIEW

File: src/components/contexts/MapContextProvider.tsx [hunk 1]
```diff
@@ -32,13 +32,7 @@ interface IMapContext {
...
   areStopsVisible: boolean;
   propertyGrid: Stop | Conversation | null;
-  connectionData: Array<{
-    coordinates: {lat: number; lon: number};
-    id: string;
-    name: string;
-    ref: string;
-    isOsm: boolean;
-  }>;
+  connectionData: Array<Stop>;
```
Comment (AdamSarach) [inline src/components/contexts/MapContextProvider.tsx]: :+1:

File: src/components/contexts/MapContextProvider.tsx [hunk 2]
```diff
@@ -248,9 +236,7 @@ const MapContextProvider: FC = ({children}) => {
   const [areStopsVisible, setAreStopsVisible] = useState(false);
   const [propertyGrid, setPropertyGrid] = useState<Stop | Conversation | null>(null);
   const [rerenderConnections, setRerenderConnections] = useState(false);
-  const [connectionData, setConnectionData] = useState<
-    Array<{coordinates: {lat: number; lon: number}; id: string; name: string; ref: string; isOsm: boolean}>
-  >([]);
+  const [connectionData, setConnectionData] = useState<Array<Stop>>([]);
```
Comment (AdamSarach) [inline src/components/contexts/MapContextProvider.tsx]: `<Array<Stop>>` is equivalent to  `<Stop[]>`. I personally like to use `Array` only when there is more code and there is possibility for square brackets to disappear among code mess. It is just a suggestion.

File: src/components/contexts/MapContextProvider.tsx [hunk 3]
```diff
@@ -301,14 +287,11 @@ const MapContextProvider: FC = ({children}) => {
...
 
-  const updateConnectionData = useCallback(
-    (data: {coordinates: {lat: number; lon: number}; id: string; name: string; ref: string; isOsm: boolean}) => {
-      if (data) {
-        setConnectionData(oldState => [...oldState, data]);
-      }
-    },
-    [],
-  );
+  const updateConnectionData = useCallback((data: Stop) => {
```
Comment (AdamSarach) [inline src/components/contexts/MapContextProvider.tsx:290]: :+1:

File: src/components/MapView.tsx [hunk 4]
```diff
@@ -98,24 +98,17 @@ export const MapView = () => {
...
-    ref: string,
-  ) => {
+  const createConnection = (busStop: Stop) => {
     if (connectionData.length < 2) {
-      const isOsm = stopType === 0;
-      const entryPoint = {coordinates, id, isOsm, name, ref};
-
-      if (connectionData.length === 1 && connectionData[0].isOsm === isOsm) {
-        if (connectionData[0].id !== id) {
+      // Check if stop is the same type as selected one
```
Comment (AdamSarach) [inline src/components/MapView.tsx:103]: Do we really need these comments? They just explain what code does, nothing more. If code is too ambiguous, how about making one-liner function like
`function sameStopClicked (connectionData) {return connectionData.length === 1 && connectionData[0].stopType === busStop.stopType) }`
Then: `if (sameStopClicked(connectionData)) { ... }`

File: src/components/ConnectionSidePanel.jsx [hunk 5]
```diff
@@ -70,26 +70,18 @@ const ConnectionSidePanel = () => {
...
-          </span>
+          <span>{connectionData.length > 0 ? generateStopName(connectionData[0]) : '...'}</span>
         </fieldset>
 
         <fieldset
           className={`connection-prompt__field ${
             error && connectionData.length < 2 && 'connection-prompt__field--error'
           }`}>
-          <legend>Wskaż drugi przystanek</legend>
+          <legend>{t('connection.chooseSecondStop')}</legend>
```
Comment (AdamSarach) [inline src/components/ConnectionSidePanel.jsx:82]: Are these translations in other PR?

**Review by AdamSarach — APPROVED** [review]
Just a few issues to discuss

File: src/components/contexts/MapContextProvider.tsx (hunk 2 above)
Comment (luktar) [reply src/components/contexts/MapContextProvider.tsx]: Stop[] is better. I'll fix it tomorrow.

File: src/components/MapView.tsx (hunk 4 above)
Comment (luktar) [reply src/components/MapView.tsx:103]: It's good suggestions but I prefer these comments - it's shorter just in this case.

File: src/components/ConnectionSidePanel.jsx (hunk 5 above)
Comment (luktar) [reply src/components/ConnectionSidePanel.jsx:82]: I haven't found any.

**Pull request merged**